src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -907,6 +907,18 @@ private void ThrowIfCannotActivateLicense(LicenseStatus newLicenseStatus)
                                        "In order to use this license please disable SNMP Monitoring in the server configuration";
                 throw GenerateLicenseLimit(LimitType.Snmp, message);
             }
+            
+            if (_serverStore.LicenseManager.LicenseStatus.Type != LicenseType.Developer && newLicenseStatus.Type == LicenseType.Developer)
+            {
+                if (_serverStore.Server.Certificate.Certificate.NotAfter > DateTime.UtcNow.AddMonths(4))
```

Comment:
For dev license, also make sure that the _total_ duration is within the limit.

Reply:
Hi, `total duration` of the license itself? Or something related to the certificate?

Reply:
For the license itself, yes.

Reply:
Move that to SecretProtection (create ValidateExpiration overload?) also let's put '4' as a const inside SecretProtection so it would be easier to pinpoint all the places that touch dev license expiration